app/Components/Sudoku Board/SudokuBoard.tsx

```diff
@@ -438,13 +440,69 @@ const SudokuBoard = (props: SudokuBoardProps) => {
     const inputValue = event.nativeEvent.key;
     if (/^[1-9]$/.test(inputValue)) {
       updateCellEntry(parseInt(inputValue, 10));
-    }
-    if (
+    } else if (
       inputValue == "Delete" ||
       inputValue == "Backspace" ||
-      inputValue == "0"
-    )
+      inputValue == "0" ||
+      inputValue == "e" ||
+      inputValue == "E" // e and E are for erase
+    ) {
       eraseSelected();
+    } else if (inputValue == "u" || inputValue == "U") {
+      undo();
+    } else if (inputValue == "p" || inputValue == "P") {
+      saveGame(sudokuBoard);
+      navigation.goBack();
+    } else if (
+      inputValue == "t" ||
+      inputValue == "T" ||
+      inputValue == "n" ||
+      inputValue == "N"
+    ) {
+      toggleNoteMode();
+    } else if (sudokuBoard.selectedCell) {
+      let newCol = sudokuBoard.selectedCell.c;
+      let newRow = sudokuBoard.selectedCell.r;
+      if (inputValue == "ArrowLeft" || inputValue == "a" || inputValue == "A") {
+        newCol--;
+        if (newCol < 0) {
+          newCol = sudokuBoard.puzzle.length - 1;
+        }
+      } else if (
+        inputValue == "ArrowRight" ||
+        inputValue == "d" ||
+        inputValue == "D"
+      ) {
+        newCol++;
+        if (newCol >= sudokuBoard.puzzle.length) {
+          newCol = 0;
+        }
+      } else if (
+        inputValue == "ArrowUp" ||
+        inputValue == "w" ||
+        inputValue == "W"
+      ) {
+        newRow--;
+        if (newRow < 0) {
+          newRow = sudokuBoard.puzzle.length - 1;
+        }
+      } else if (
+        inputValue == "ArrowDown" ||
+        inputValue == "s" ||
+        inputValue == "S"
+      ) {
+        newRow++;
+        if (newRow >= sudokuBoard.puzzle.length) {
+          newRow = 0;
+        }
+      } else {
```

Comment:
We can simplify this logic a little using mathematical modulo (Javascript modulo won't work since it has different rules ☹️ )

```typescript
      if (inputValue == "ArrowLeft" || inputValue == "a" || inputValue == "A") {
        newCol = mathimaticalMod(--newCol, sudokuBoard.puzzle.length);
      } else if (
        inputValue == "ArrowRight" ||
        inputValue == "d" ||
        inputValue == "D"
      ) {
        newCol = mathimaticalMod(++newCol, sudokuBoard.puzzle.length);
      } else if (
        inputValue == "ArrowUp" ||
        inputValue == "w" ||
        inputValue == "W"
      ) {
        newRow = mathimaticalMod(--newRow, sudokuBoard.puzzle.length);
      } else if (
        inputValue == "ArrowDown" ||
        inputValue == "s" ||
        inputValue == "S"
      ) {
        newRow = mathimaticalMod(++newRow, sudokuBoard.puzzle.length);
      } else {
        return;
      }
```
And the `mathimaticalMod` function:
```typescript
  /**
   * Javascript mod function does not preform mathimatical modulo operation
   * See article for details https://web.archive.org/web/20090717035140if_/javascript.about.com/od/problemsolving/a/modulobug.htm
   * @param a The dividend
   * @param n The divisor
   * @returns The result of the mod operation
   */
  const mathimaticalMod = (a: number, n: number) => {
    return ((a % n) + n) % n;
  };
```

Reply:
Good insight to use a function to simplify this math and that the modulus operator alone would not work! 

I didn't realize that when it comes to a single negative number in a modulus operation there is disagreement between programming languages over what the correct answer should be.

I tried your function but it doesn't appear to work for this use case. For instance, consider the very first case where someone enter arrow left and gets to -1 (from 0 on range 0 to 8). We want them to loop back around to the maximum value of 8. Your function evaluates in js as follows:
```
((-1 % 8) + 8) % 8
((-1)+8)%8
7%8
```
Which results in 7 not the desired 8.

That did lead me to realize that what we actually want is a wrapping function like the following which I will implement in code:
```js
function wrapNumber(number, lowerBound, upperBound) {
  const range = upperBound - lowerBound + 1; // Calculate the total range
  return (number + range) % range + lowerBound; // Wrap around using modulo
}
```
Since modulus gets the remainder after dividing the follow section of code: `(number + range) % range` is equivalent to just `number % range` for positive numbers as the added range will be stripped out as evenly divisible. Since modulus already worked for wrapping for positive numbers that is desired. For negative numbers, however, it works by simply making them positive. For instance, `(-1 + 9) % 9 = 8` as desired.

Reply:
Nice to break the functions outside of SudokuBoard.tsx. The code suggestion I made works with n = 9 which is the length of the sudoku board. But your functions are equivalent!